### PR TITLE
Add openAPI documentation

### PR DESCRIPTION
**Pull request by whytheplatypus**
- [x] /regulations*
- [x] /layer* (further work may be needed)
- [x] /diff*
- [x] /notice/*
- [x] /preamble/*
- [x] /search*

### CODE REVIEW

**Comment by whytheplatypus** [comment]
Rendered version at: https://petstore.swagger.io/?url=https://raw.githubusercontent.com/CMSgov/regulations-core/openapi-docs/openapi.yaml

File: openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,1367 @@
...
+      summary: "Return a Notice"
+      description: "Return a Notice and a List of its Amendments"
+      responses:
+        "200":
+          description: "Successful operation"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Notice"
+ /preamble/{label_id}:
```
Comment (whytheplatypus) [inline openapi.yaml]: correct indent.
```suggestion
  /preamble/{label_id}:
```

File: openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,1367 @@
...
+                  $ref: "#/components/schemas/NoticeList"
+          "400":
+            description: "Invalid value"
+  /notice/{docnum}:
+    get:
+      parameters:
+        - name: "docnum"
+          in: "path"
+          summary: "The document number for the notice"
+          description: "The document number for the notice"
```
Comment (whytheplatypus) [inline openapi.yaml]: apparently summary isn't a think for a parameter
```suggestion
          description: "The document number for the notice"
```

**Review by theabrad — APPROVED** [review]

**Review by brittag — APPROVED** [review]
Looks good! It'll be cool to be able to flesh this out more as we get further, like if we figure out more about what the preamble stuff can do.

**Pull request merged**